pageserver/src/remote_storage/local_fs.rs

```diff
@@ -154,6 +153,58 @@ impl RemoteStorage for LocalFs {
         }
     }
 
+    async fn download_range(
+        &self,
+        from: &Self::StoragePath,
+        start_inclusive: u64,
+        end_inclusive: Option<u64>,
+        to: &mut (impl io::AsyncWrite + Unpin + Send + Sync),
+    ) -> anyhow::Result<()> {
+        if Some(start_inclusive) == end_inclusive {
+            return Ok(());
+        }
+        ensure!(
+            start_inclusive < end_inclusive.unwrap_or(u64::MAX),
+            "Invalid range, start ({}) is bigger then end ({:?})",
+            start_inclusive,
+            end_inclusive
+        );
```

Comment:
Would it make more sense to put both of these checks inside if let Some(end_inclusive) = end_inclusive { ... }